src/tool_operate.c

```diff
@@ -1736,9 +1736,9 @@ static CURLcode single_transfer(struct GlobalConfig *global,
             my_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
           }
           else {
-            my_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);
-            /* libcurl default is strict verifyhost -> 2L   */
-            /* my_setopt(curl, CURLOPT_SSL_VERIFYHOST, 2L); */
+            /* libcurl default is strict verifyhost -> 1L, verifypeer -> 1L */
+            /* my_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L); */
+            /* my_setopt(curl, CURLOPT_SSL_VERIFYHOST, 1L); */
```

Comment:
Speaking of the most important security settings for TLS to actually work, wouldn't it better to keep them here for clarity and making sure we don't rely on defaults set elsewhere in the code (often residing in a separate shared libcurl library)?

The the DOH/proxy insecure checks also don't enable security explicitly, or do it only for `VERIFYPEER`. IMO correcting those to set both to `1L` would be a better direction for robustness.

Reply:
[lib]curl was "secure by default" since ever, and this is extremely unlikely to change in future versions, and `VERIFYHOST` is already excluded. Digging through history, I found 0417d34533c187db1ac0fadd6765573defb69d00 by @bagder, which is quite similar to this change.

Reply:
It was done to make `--libcurl` output easier to follow, in 2010. I'd vote for being explicit when it comes to security, but if this is just me, the `my_setopt(curl, CURLOPT_PROXY_SSL_VERIFYPEER, 1L);` line shouldn't also go?

Reply:
It's commented out, similar to `VERIFYHOST` (for `VERIFYHOST` I only changed the default to 1, see my previous comment).

I don't have a strong opinion on this, will let @bagder decide.